crates/sui-light-client/src/mmr.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use fastcrypto::hash::{Blake2b256, HashFunction};
+use fastcrypto::merkle::MerkleTree;
+use move_core_types::u256::U256;
+use sui_types::accumulator_root::{EventCommitment, EventStreamHead};
+
+const U256_ZERO: U256 = U256::zero();
+
+fn hash_two_to_one_u256(left: U256, right: U256) -> U256 {
+    let mut concatenated = bcs::to_bytes(&left).expect("Failed to serialize left U256");
+    concatenated.extend_from_slice(&bcs::to_bytes(&right).expect("Failed to serialize right U256"));
+    let hash = Blake2b256::digest(&concatenated);
+    U256::from_le_bytes(&hash.digest)
+}
+
+// Build the Merkle root of all events in a single checkpoint
+fn build_event_merkle_root(events: &[EventCommitment]) -> U256 {
+    // Debug assertion to ensure events are ordered by the natural order of EventCommitment
+    debug_assert!(
+        events.windows(2).all(|pair| {
+            let (a, b) = (&pair[0], &pair[1]);
+            (a.checkpoint_seq, a.transaction_idx, a.event_idx)
+                <= (b.checkpoint_seq, b.transaction_idx, b.event_idx)
```

Comment:
can we just `impl Ord, PartialOrd for EventCommitment` and compare the pairs directly?